uspace/lib/c/generic/thread/fibril.c

```diff
@@ -795,7 +795,8 @@ int fibril_test_spawn_runners(int n)
 
 	if (!multithreaded) {
 		_ready_debug_check();
-		futex_initialize(&ready_semaphore, ready_st_count);
+		if (futex_initialize(&ready_semaphore, ready_st_count) != EOK)
+			abort();
```

Comment:
Random idea: What if instead of aborting on failure, we fall back to a spinlock+thread_yield()?